libraries/src/Component/Router/Rules/PreprocessRules.php

```diff
@@ -0,0 +1,169 @@
+<?php
+
+/**
+ * Joomla! Content Management System
+ *
+ * @copyright  (C) 2024 Open Source Matters, Inc. <https://www.joomla.org>
+ * @license    GNU General Public License version 2 or later; see LICENSE.txt
+ */
+
+namespace Joomla\CMS\Component\Router\Rules;
+
+use Joomla\CMS\Component\Router\RouterViewConfiguration;
+use Joomla\Database\DatabaseAwareTrait;
+use Joomla\Database\ParameterType;
+
+// phpcs:disable PSR1.Files.SideEffects
+\defined('_JEXEC') or die;
+// phpcs:enable PSR1.Files.SideEffects
+
+/**
+ * Rule to prepare the query and add missing information
+ *
+ * This rule adds the alias to an ID query parameter and the
+ * category ID if either of them is missing. This requires that
+ * the db table contains an alias column.
+ * This fixes sloppy URLs in the code, but doesn't mean you can
+ * simply drop the alias from the &id= in the future. Cleaning up
+ * every request with this would mean a significant performance impact
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class PreprocessRules implements RulesInterface
+{
+    use DatabaseAwareTrait;
+
+    /**
+     * View to prepare
+     *
+     * @var   RouterViewConfiguration
+     * @since __DEPLOY_VERSION__
+     */
+    protected $view;
+
+    /**
+     * DB Table to read the information from
+     *
+     * @var   string
+     * @since __DEPLOY_VERSION__
+     */
+    protected $table;
+
+    /**
+     * ID column in the table to read the information from
+     *
+     * @var   string
+     * @since __DEPLOY_VERSION__
+     */
+    protected $key;
+
+    /**
+     * Parent ID column in the table to read the information from
+     *
+     * @var   string
+     * @since __DEPLOY_VERSION__
+     */
+    protected $parent_key;
+
+    /**
+     * Class constructor.
+     *
+     * @param   RouterViewConfiguration  $view        View to act on
+     * @param   string                   $table       Table name for the views information
+     * @param   string                   $key         Key in the table to get the information
+     * @param   string                   $parent_key  Column name of the parent key
+     *
+     * @since   __DEPLOY_VERSION__
+     */
+    public function __construct(RouterViewConfiguration $view, $table, $key, $parent_key = null)
+    {
+        $this->view       = $view;
+        $this->table      = $table;
+        $this->key        = $key;
+        $this->parent_key = $parent_key;
+    }
+
+    /**
+     * Finds the right Itemid for this query
```

Comment:
```suggestion
     * Finds the correct Itemid for this query
```

less ambiguous english